mlx/backend/metal/utils.h

```diff
@@ -123,6 +124,31 @@ MTL::Size get_block_dims(int dim0, int dim1, int dim2) {
   return MTL::Size{1ul << pows[0], 1ul << pows[1], 1ul << pows[2]};
 }
 
+NS::String* make_string(std::ostringstream& os) {
+  return NS::String::string(os.str().c_str(), NS::UTF8StringEncoding);
+}
+
+void debug_set_stream_queue_label(MTL::CommandQueue* queue, int index) {
+#ifdef MLX_METAL_DEBUG
+  std::ostringstream label;
+  label << "Stream " << index;
+  queue->setLabel(make_string(label));
+#endif
+}
+
+void debug_set_primitive_buffer_label(Primitive& primitive) {
+#ifdef MLX_METAL_DEBUG
```

Comment:
I would mark these explicitly inline.. I don't think apple clang will do it by default (you will get duplicate linker errors if you use them in another object file).